agenda/presentation/src/screenshotTest/kotlin/com/pronoidsoftware/agenda/presentation/detail/reminder/ReminderDetailScreenPreviews.kt

```diff
@@ -1,73 +1,91 @@
 package com.pronoidsoftware.agenda.presentation.detail.reminder
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
 import androidx.compose.ui.tooling.preview.Preview
+import com.pronoidsoftware.core.presentation.designsystem.LocalClock
 import com.pronoidsoftware.core.presentation.designsystem.TaskyTheme
+import com.pronoidsoftware.testutil.jvmtest.core.data.time.TestClock
 import kotlinx.datetime.LocalDate
 import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+
+private val selectedDate = LocalDate(2022, 3, 1)
+private val selectedNotificationTime = LocalDateTime(2022, 7, 21, 8, 0)
+private val today = LocalDate(2024, 7, 19)
+private val fixedClock = TestClock(today.atStartOfDayIn(TimeZone.currentSystemDefault()))
 
 @Preview
 @Composable
 private fun ReminderDetailScreenPreview_Read() {
     TaskyTheme {
-        ReminderDetailScreen(
-            state = ReminderDetailState(
-                title = "Project X",
-                description = "Weekly plan\nRole distribution",
-                selectedDate = LocalDate(2022, 3, 1),
-                atTime = LocalDateTime(2022, 7, 21, 8, 0),
-                isEditing = false,
-            ),
-            onAction = {},
-        )
+        CompositionLocalProvider(LocalClock provides fixedClock) {
+            ReminderDetailScreen(
+                state = ReminderDetailState(
+                    title = "Project X",
+                    description = "Weekly plan\nRole distribution",
+                    selectedDate = selectedDate,
+                    atTime = selectedNotificationTime,
+                    isEditing = false,
+                ),
+                onAction = {},
+            )
+        }
     }
 }
 
 @Preview
 @Composable
 private fun ReminderDetailScreenPreview_Read_EmptyDescription() {
     TaskyTheme {
-        ReminderDetailScreen(
-            state = ReminderDetailState(
-                title = "Project X",
-                selectedDate = LocalDate(2022, 3, 1),
-                atTime = LocalDateTime(2022, 7, 21, 8, 0),
-                isEditing = false,
-            ),
-            onAction = {},
-        )
+        CompositionLocalProvider(LocalClock provides fixedClock) {
```

Comment:
Is this fixed clock a custom clock? Or is it the default one? Would there be any issues with using the default clock for the preview? Then you wouldn't need to declare this for all previews.

Reply:
Unfortunately yes, there would be an issue with just using the default clock here because these previews are for screenshot testing. I use the fixed clock so that I know that the time will be exactly the same each time instead of updating to the current time, which work break each screenshot test. I supposed there's nothing within this file itself that denotes that these are for screenshot testing, but it is located in the screenshotTest source set, which is easier to see within an IDE

Reply:
Good point, makes sense 👍